### PR TITLE
Feat/device driver

### PR DESCRIPTION
**Pull request by yashrahmed**
Code for Controller Firmware + Java interface for the device

### CODE REVIEW

File: plotting-app/hid-driver/pom.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>plotting-app</artifactId>
+        <groupId>org.unknown.plotting</groupId>
```
Comment (anujparikh) [inline plotting-app/hid-driver/pom.xml:7]: did we miss out to name package properly?

File: plotting-app/hid-driver/src/main/java/org/unknown/plottingapp/hiddriver/models/HIDState.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                "leftLR=" + leftLR +
+                ", leftUD=" + leftUD +
+                ", rightLR=" + rightLR +
+                ", rightUD=" + rightUD +
+                '}';
+    }
+
+
+    private StickPosition getHorizontal(int yPos) {
+        if (yPos > 700) return StickPosition.RIGHT;
```
Comment (anujparikh) [inline plotting-app/hid-driver/src/main/java/org/unknown/plottingapp/hiddriver/models/HIDState.java:51]: how did we come up with these number? (700, 200, ..)

File: plotting-app/pom.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (anujparikh) [inline plotting-app/pom.xml:1]: why we have 2 different `pom.xml` files?

**Review by anujparikh — APPROVED** [review]
Overall looks good. Couple of doubts from my side.

File: plotting-app/hid-driver/pom.xml (hunk 1 above)
Comment (yashrahmed) [reply plotting-app/hid-driver/pom.xml:7]: Not really. That's the name I went with for now. We could change it later.

File: plotting-app/hid-driver/src/main/java/org/unknown/plottingapp/hiddriver/models/HIDState.java (hunk 2 above)
Comment (yashrahmed) [reply plotting-app/hid-driver/src/main/java/org/unknown/plottingapp/hiddriver/models/HIDState.java:51]: The device is an analog device that uses an ADC to map signals to a range of [0,1024)

File: plotting-app/pom.xml (hunk 3 above)
Comment (yashrahmed) [reply plotting-app/pom.xml:1]: We have an outer module, and an inner module (the one used for the HiDDriver module). We will have more in the future.

**Pull request merged**